### PR TITLE
Wrap string condition expressions in apostrophes automatically

### PR DESCRIPTION
**Pull request by thomhurst**
StringCondition class for wrapping strings in quotes.

This class is to be inherited by String-type conditions such as Contains, StartsWith and EndsWith.

See issue #227

### CODE REVIEW

File: src/Sharpliner/AzureDevOps/ConditionedExpressions/Condition.cs [hunk 1]
```diff
@@ -66,6 +69,17 @@ public static string RemoveTags(string condition)
         return condition;
     }
 
+    protected static string WrapQuotes(string value)
```
Comment (premun) [inline src/Sharpliner/AzureDevOps/ConditionedExpressions/Condition.cs:72]: Can you use this method in `BranchCondition` too?

File: tests/Sharpliner.Tests/AzureDevOps/ConditionedExpressions/ConditionalsTests.cs [hunk 2]
```diff
@@ -271,4 +271,85 @@ public void Custom_Condition_Test()
...
+    {
+        public override Pipeline Pipeline => new()
+        {
+            Variables =
+            {
+                If.Contains("refs/heads/feature/", variables.Build.SourceBranch)
+                    .Variable("feature", "on"),
+
+                If.Contains("refs/heads/feature/", variables["Build.SourceBranch"])
+                .Variable("feature", "on")
```
Comment (premun) [inline tests/Sharpliner.Tests/AzureDevOps/ConditionedExpressions/ConditionalsTests.cs:285]:
```suggestion
                    .Variable("feature", "on")
```

**Review by premun — COMMENTED** [review]
I think this is a nice improvement! I was trying to think of a case where we would force these and force the YAML to break with this but cannot think of any.

It's a breaking change, but hopefully that's okay. I will bump the minor version number.

Left just two minor comments, thanks!

File: src/Sharpliner/AzureDevOps/ConditionedExpressions/Condition.cs (hunk 1 above)
Comment (thomhurst) [reply src/Sharpliner/AzureDevOps/ConditionedExpressions/Condition.cs:72]: BranchCondition inherits from EqualityCondition. I made that base class Inherit from String condition. 
What do you think?

It surfaced some existing tests that were asserting an incorrect syntax, so I've fixed those now too.

File: tests/Sharpliner.Tests/AzureDevOps/ConditionedExpressions/ConditionalsTests.cs (hunk 2 above)
Comment (thomhurst) [reply tests/Sharpliner.Tests/AzureDevOps/ConditionedExpressions/ConditionalsTests.cs]: Fixed :)

File: src/Sharpliner/AzureDevOps/ConditionedExpressions/Condition.cs (hunk 1 above)
Comment (premun) [reply src/Sharpliner/AzureDevOps/ConditionedExpressions/Condition.cs:72]: Great, that is the best outcome

File: tests/Sharpliner.Tests/AzureDevOps/ConditionedExpressions/ConditionalsTests.cs [hunk 3]
```diff
@@ -31,7 +31,7 @@ public void And_Condition_Test()
         variable.Condition!.ToString().Should().Be(
             "and(" +
                 "notIn('bar', 'foo', 'xyz', 'foo'), " +
-                "ne($(Configuration), 'Debug'), " +
+                "ne('$(Configuration)', 'Debug'), " +
```
Comment (thomhurst) [inline tests/Sharpliner.Tests/AzureDevOps/ConditionedExpressions/ConditionalsTests.cs:34]: These tests were asserting an incorrect syntax. I made EqualityCondition inherit from StringCondition and it fixed this incorrect syntax, so I've updated all these tests to work now.

Comment (premun) [reply tests/Sharpliner.Tests/AzureDevOps/ConditionedExpressions/ConditionalsTests.cs:34]: I can see there are some failing around quoting `boolean` values. Do you know what's the correct syntax in YAML? Does it matter if you specify variable is `true` and compare with `'true'`?

Comment (premun) [reply tests/Sharpliner.Tests/AzureDevOps/ConditionedExpressions/ConditionalsTests.cs:34]: Thanks for trying it out!

**Review by premun — APPROVED** [review]

**Pull request merged**